File: javascript/49-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
  let hashMap = {};

  for(let i = 0; i < strs.length; i++) {
      let str = strs[i].split("").sort().join("");

      if (str in hashMap) {
          hashMap[str].push(strs[i]);
      } else {
          hashMap[str] = [strs[i]];
      }
  }

  return Object.values(hashMap);
};

File: python/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}

        for string in strs:
            sortedString = list(string)
            sortedString.sort()
            sortedString = "".join(sortedString)

            if sortedString in dict:
                dict[sortedString].append(string)
            else:
                dict[sortedString] = [string]

        return list(dict.values())